### PR TITLE
Instrument handwritten .wat files

### LINKED ISSUE #27: Failed default output_path becuase parent directory doesn't exists
If I don't have the `output` directory, running `cargo run -- vis-script --script tests/scripts/fault_injection/dfinity_sync.mm` will give error

```
Error: Could not open "<OMITTED>/whamm-fork/output/vis.svg" for writing : No such file or directory
```

We can create the parent directory if it doesn't exist and reduce the assumption for user by putting `std::fs::create_dir_all` somewhere

### PR DESCRIPTION
**Pull request by ahuoguo**
Added some integration testing for handwritten `.wat` files and `wasm` files generated from rust

part of the issue is handwritten .wat files will lose the function's name when we run wat2wasm on it. However, their name is preserved if it's emitted from cargo build --target wasm32-unknown-unknown

Also `target_imp_name` seems to be the wrong name.

Also fixes #27

### CODE REVIEW

File: tests/apps/handwritten/add.wasm
Comment (ejrgilbert) [inline tests/apps/handwritten/add.wasm:1]: I would only check in the `.wat` file, not the `.wasm`, just reduces the size of the repo.

File: tests/integration_test.rs [hunk 1]
```diff
@@ -94,6 +95,72 @@ fn instrument_dfinity_with_fault_injection() {
...
+    assert!(res.status.success());
+
+    let out = Command::new("wasm2wat")
+        .arg("output/output.wasm")
+        .output()
+        .expect("failed to execute process");
+
+    // write to file
+    let stdout = String::from_utf8(out.stdout).unwrap();
+    fs::write("output/output.wat", stdout).unwrap();
```
Comment (ejrgilbert) [inline tests/integration_test.rs:161]: You wouldn't need to write this to a file, we've already done assertions at this point. All we care about is the `expect` above. Also, if you have `wabt` included as a dependency, is there a `wasm2wat` utility we can use instead? Like you used `wat2wasm` above?

File: tests/scripts/instr.mm [hunk 2]
```diff
@@ -0,0 +1,10 @@
+i32 i;
+wasm:bytecode:call:before /
+    // true
+    // &&
+    // target_fn_type == "local" &&
```
Comment (ejrgilbert) [inline tests/scripts/instr.mm:5]: remove commented out code.

**Review by ejrgilbert — CHANGES_REQUESTED** [review]

File: tests/integration_test.rs (hunk 1 above)
Comment (ahuoguo) [reply tests/integration_test.rs:161]: I was writing to file for easier manual inspection of the instrumentation behavior.

**Review by ejrgilbert — APPROVED** [review]

**Pull request merged**